qiskit/visualization/text.py

```diff
@@ -954,8 +954,10 @@ def add_connected_gate(instruction, gates, layer, current_cons):
                 return layer, current_cons, connection_label
 
             for qubit in instruction.qargs:
-                layer.set_qubit(qubit, Barrier())
-
+                try:
+                    layer.set_qubit(qubit, Barrier())
+                except ValueError:
+                    dummy = 0
```

Comment:
This seems like a very hacky way to ignore an exception. When possible it's better to avoid the exception catch. Let me suggest this:
```suggestion
                if qubit in self.qregs:
                    layer.set_qubit(qubit, Barrier())
```